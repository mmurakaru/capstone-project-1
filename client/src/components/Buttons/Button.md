Save button

```jsx
<Button onClick={() => {}} style={{ width: '60px', height: '60px' }}>
  <img src={'./src/assets/styleguide/save-green.svg'} />
</Button>
```

Save button disabled

```jsx
<Button disabled onClick={() => {}} style={{ width: '60px', height: '60px' }}>
  <img src={'./src/assets/styleguide/save-green-disabled.svg'} />
</Button>
```

Delete button

```jsx
<Button
  isDeleteButton
  onClick={() => {}}
  style={{ width: '60px', height: '60px' }}
>
  <img src={'./src/assets/styleguide/x-red.svg'} />
</Button>
```
