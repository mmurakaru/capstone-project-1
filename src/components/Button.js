import styled from 'styled-components/macro'

const Button = styled.button`
  background-color: transparent;
  border: var(--primary-dark) 2px solid;
  color: var(--primary-dark);
  padding: 6px;
  font-size: 0.8em;
  width: 100%;

  &:disabled {
    opacity: 50%;
  }
`
export default Button
