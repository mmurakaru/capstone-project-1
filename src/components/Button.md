Regular Button

```js
<Button>Save this song</Button>
```

```js
<Button className="active">Remove this song</Button>
```
