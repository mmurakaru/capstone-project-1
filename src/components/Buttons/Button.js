import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

Button.propTypes = {
  disabled: PropTypes.bool.isRequired,
  handleClick: PropTypes.func,
  children: PropTypes.string.isRequired,
}

export default function Button({ disabled, handleClick, children }) {
  return (
    <ButtonStyled
      disabled={disabled}
      onClick={handleClick}
      data-testid="button"
    >
      {children}
    </ButtonStyled>
  )
}

const ButtonStyled = styled.button`
  background-color: transparent;
  border: var(--primary-dark) 2px solid;
  color: var(--primary-dark);
  padding: 6px;
  font-size: 0.8em;
  width: 100%;

  &:disabled {
    opacity: 50%;
  }
`
