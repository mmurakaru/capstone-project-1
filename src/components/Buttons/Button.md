Regular Button

```js
<Button disabled={false} handleClick={() => {}}>
  Save this song
</Button>
```

```js
<Button disabled={true} handleClick={() => {}}>
  Save this song
</Button>
```
