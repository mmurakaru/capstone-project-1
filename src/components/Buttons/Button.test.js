import { render } from '@testing-library/react'
import user from '@testing-library/user-event'
import Button from './Button'

describe('Button', () => {
  it('renders correctly', () => {
    const { container, rerender } = render(
      <Button disabled={false} handleClick={() => {}}>
        Save this song
      </Button>
    )
    expect(container.firstChild).toMatchSnapshot()

    rerender(
      <Button disabled={true} handleClick={() => {}}>
        Save this song
      </Button>
    )
    expect(container.firstChild).toMatchSnapshot()
  })
  it('calls handleClick with the correct id', () => {
    const handleClickMock = jest.fn()
    const { getByRole } = render(
      <Button disabled={false} handleClick={() => handleClickMock('a')}>
        Save this song
      </Button>
    )
    const button = getByRole('button')
    user.click(button)

    expect(handleClickMock).toHaveBeenCalledWith('a')
  })
})
