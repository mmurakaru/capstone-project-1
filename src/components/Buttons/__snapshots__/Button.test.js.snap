// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
.c0 {
  background-color: transparent;
  border: var(--primary-dark) 2px solid;
  color: var(--primary-dark);
  font-size: 0.8em;
  padding: 6px;
  width: 100%;
}

.c0:disabled {
  opacity: 50%;
}

<button
  class="c0"
  data-testid="button"
>
  Save this song
</button>
`;

exports[`Button renders correctly 2`] = `
<button
  class="Button__ButtonStyled-o1fnpc-0 eSgmRv"
  data-testid="button"
  disabled=""
>
  Save this song
</button>
`;
